h2o-automl/src/main/java/ai/h2o/automl/preprocessing/TargetEncoding.java

```diff
@@ -0,0 +1,179 @@
+package ai.h2o.automl.preprocessing;
+
+import ai.h2o.automl.AutoML;
+import ai.h2o.targetencoding.TargetEncoder;
+import ai.h2o.targetencoding.TargetEncoderModel;
+import ai.h2o.targetencoding.TargetEncoderModel.DataLeakageHandlingStrategy;
+import ai.h2o.targetencoding.TargetEncoderModel.TargetEncoderParameters;
+import ai.h2o.targetencoding.TargetEncoderPreprocessor;
+import hex.Model;
+import hex.Model.Parameters.FoldAssignmentScheme;
+import hex.ModelPreprocessor;
+import water.DKV;
+import water.Iced;
+import water.Key;
+import water.fvec.Frame;
+import water.fvec.Vec;
+import water.rapids.ast.prims.advmath.AstKFold;
+import water.util.ArrayUtils;
+
+import java.util.*;
+
+public class TargetEncoding implements PreprocessingStep {
+    
+    static String TE_FOLD_COLUMN_SUFFIX = "_te_fold";
+    
+    private AutoML _aml;
+    private TargetEncoderPreprocessor _tePreprocessor;
+    private TargetEncoderModel _teModel;
+    private final List<Completer> _disposables = new ArrayList<>();
+    private int _cardinalityThreshold = 0;  // the minimal cardinality for a column to be TE encoded. 
```

Comment:
@ledell this one's for you :)

Reply:
@sebhrusen Thank you!

Reply:
@goldentom42 if you want to change the cardinality, it's here